djedi/admin/mixins.py

```diff
@@ -38,5 +40,6 @@ def get_context_data(self, **context):
 
         context['THEME'] = theme
         context['VERSION'] = djedi.__version__
+        context['PLUGINS'] = mark_safe(list(plugins.plugins.keys()))
```

Comment:
We can json serialize keys here to ensure that the list of plugin extensions passed is safe to load up in js.

Something like:

```
context['PLUGINS'] = mark_safe(json.dumps(list(plugins.plugins.keys())))
```

Reply:
context isn't necessarily loaded by js, as shown by
```
        if request.is_ajax():
            return self.render_to_json(context)
        else:
            return self.render_plugin(request, context)
```
in `admin/api.py`

Reply:
Hmm, seems like it's either returned as JSON or used in template context to render the list plugin editor, or maybe I'm missing something? From what I can see in the https://github.com/5monkeys/djedi-cms/blob/ef5191e8501a840e5f05dd8d0f071011e0ad437a/djedi/templates/djedi/plugins/list/editor.html#L31, `PLUGINS` is passed as a config to init ListEditor? I tested with Python 2 and the extension string for some plugins ends up with a unicode `u`  for example, which breaks loading up the editor.

Reply:
Ah,  that's no good. High time for some frontend tests as well for these cases. Would it be possible to json encode it on the template itself maybe? That way if any plugin needs to use the `PLUGINS` context data there's no need for decoding.

Reply:
`json.dumps` added to plugin list, turns out that was the best solution